src/Tools/dotnet-user-jwts/src/Helpers/DevJwtCliHelpers.cs

```diff
@@ -0,0 +1,166 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.IdentityModel.Tokens.Jwt;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text.Json;
+using System.Xml.Linq;
+using System.Xml.XPath;
+using Microsoft.Extensions.CommandLineUtils;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Configuration.UserSecrets;
+
+namespace Microsoft.AspNetCore.Authentication.JwtBearer.Tools;
+
+internal static class DevJwtCliHelpers
+{
+    public static string GetUserSecretsId(string projectFilePath)
+    {
+        var projectDocument = XDocument.Load(projectFilePath, LoadOptions.PreserveWhitespace);
+        var existingUserSecretsId = projectDocument.XPathSelectElements("//UserSecretsId").FirstOrDefault();
+
+        if (existingUserSecretsId == null)
+        {
+            return null;
+        }
+
+        return existingUserSecretsId.Value;
+    }
+
+    public static string GetProject(string projectPath = null)
+    {
+        if (projectPath is not null)
+        {
+            return projectPath;
+        }
+
+        var csprojFiles = Directory.GetFiles(Directory.GetCurrentDirectory(), "*.csproj");
+        if (csprojFiles is [var path])
```

Comment:
What is this syntax o.O

Reply:
https://docs.microsoft.com/en-us/dotnet/csharp/fundamentals/functional/pattern-matching#list-patterns
https://devblogs.microsoft.com/dotnet/early-peek-at-csharp-11-features/#c-11-preview-list-patterns

List patterns are super cool. Getting closer to F# by the day:

```c#
int Sum(int[] values) => values switch
{
    [var x, .. var xs] => x + Sum(xs),
    [] => 0
};
```

Obviously, that's the most efficient way to sum an array 😉